bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/util/mapper/ActionInputsToConfigDescriptionParameters.java

```diff
@@ -0,0 +1,167 @@
+        ConfigDescriptionParameter.Type parameterType = ConfigDescriptionParameter.Type.TEXT;
+        String defaultValue = null;
+        Unit<?> unit = null;
+        boolean required = false;
+        String context = null;
+        Matcher matcher = QUANTITY_TYPE_PATTERN.matcher(input.getType());
+        if (matcher.matches()) {
+            parameterType = ConfigDescriptionParameter.Type.DECIMAL;
+            try {
+                unit = getDefaultUnit(matcher.group("dimension"), unitProvider);
+            } catch (IllegalArgumentException e) {
+                supported = false;
+            }
+        } else {
+            switch (input.getType()) {
+                case "boolean":
+                    defaultValue = "false";
+                    required = true;
+                case "java.lang.Boolean":
+                    parameterType = ConfigDescriptionParameter.Type.BOOLEAN;
+                    break;
+                case "byte":
+                case "short":
+                case "int":
+                case "long":
+                    defaultValue = "0";
+                    required = true;
+                case "java.lang.Byte":
+                case "java.lang.Short":
+                case "java.lang.Integer":
+                case "java.lang.Long":
+                    parameterType = ConfigDescriptionParameter.Type.INTEGER;
+                    break;
+                case "float":
+                case "double":
+                    defaultValue = "0";
+                    required = true;
+                case "java.lang.Float":
+                case "java.lang.Double":
+                    parameterType = ConfigDescriptionParameter.Type.DECIMAL;
+                    break;
+                case "java.lang.String":
+                    break;
+                case "java.time.LocalDate":
+                    context = "date";
+                    break;
+                case "java.time.LocalTime":
+                    context = "time";
+                    break;
+                case "java.time.LocalDateTime":
+                case "java.time.ZonedDateTime":
+                    context = "datetime";
+                    break;
+                case "org.openhab.core.library.types.DecimalType":
+                    parameterType = ConfigDescriptionParameter.Type.DECIMAL;
+                    break;
+                default:
+                    supported = false;
+                    break;
+            }
+        }
+        if (!supported) {
+            LOGGER.debug("Input parameter '{}' with type {} cannot be converted into a config description parameter!",
+                    input.getName(), input.getType());
+            return null;
+        }
+
+        ConfigDescriptionParameterBuilder builder = ConfigDescriptionParameterBuilder
+                .create(input.getName(), parameterType).withLabel(input.getLabel())
+                .withDescription(input.getDescription()).withReadOnly(false)
```

Comment:
As we decided to handle QuantityTypes by letting the user enter a decimal and assuming it is in the default unit, we have to add the unit we expect to the description (so the users knows in which unit the entered value is).
I would add the unit in brackets.

Adding the unit here is much easier than having the UI check for the unitLabel field in the config description of all decimal parameters ...

Reply:
unit is unfortunately restricted to a certain list of units and °C is even not included for example.

Reply:
Here is where the check is done:
https://github.com/openhab/openhab-core/blob/c7955e9577ad6297f7f740bfbc34921e32c06b3c/bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/ConfigDescriptionParameter.java#L185

To be analyzed if there would be any problem if we remove this check.

Reply:
We could have a new parameter in constructor  to ignore this check I believe.

Reply:
Implemented.